DuckDuckGo/Subscription/DefaultSubscriptionManager+AccountManagerKeychainAccessDelegate.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  DefaultSubscriptionManager+AccountManagerKeychainAccessDelegate.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+import Subscription
+
+extension DefaultSubscriptionManager: AccountManagerKeychainAccessDelegate {
+
+    public func accountManagerKeychainAccessFailed(accessType: AccountKeychainAccessType, error: AccountKeychainAccessError) {
+        Pixel.fire(.privacyProKeychainAccessError, withAdditionalParameters: [
+            PixelParameters.privacyProKeychainAccessType: accessType.rawValue,
+            PixelParameters.privacyProKeychainError: error.errorDescription])
```

Comment:
One possible issue here is that we won't be able to fully determine the scope of an issue, which has been an issue for Keychain errors in the past. One user can get into a bad state which then causes a stream of failures to roll in.

With the `PixelKit` API we have the ability to set a `dailyAndCount` frequency type, which can tell us the number of users affected vs. number of total events.